src/services/itemMembership/repository.ts

```diff
@@ -143,9 +145,29 @@ export const ItemMembershipRepository = AppDataSource.getRepository(ItemMembersh
         return 'item.path =' + subQuery.getQuery();
       });
 
-    if (name) {
-      query.andWhere("LOWER(item.name) LIKE '%' || :name || '%'", {
-        name: name.toLowerCase().trim(),
+    // todo: remove name
+    const allKeywords = [name, ...keywords].filter((s) => s && s.length);
```

Comment:
If you want to keep defined none empty string, this should work:

```suggestion
    const allKeywords = [name, ...keywords].filter((s) => Boolean(s));
```

But of course, your code is fine too !